devtools/running_packages.rst

```diff
@@ -114,6 +113,227 @@ with:
 
     Read more about the :ref:`deploy() <method_deploy>` method.
 
+.. _using_deploy_generator:
+
+Using deploy generator
+----------------------
+
+With the help of so-called :ref:`deploy generator <deploy_generator>`, it's possible to have all dependencies of the application to be copied into the 
+a single place for the later repackaging into the desired distribution format.
+For instance, if the application depends on boost, we may not know that it also may require many other 3rt-party libraries, 
+such as 
+`zlib <https://zlib.net/>`_, 
+`bzip2 <https://sourceware.org/bzip2/>`_, 
+`lzma <https://tukaani.org/xz/>`_, 
+`zstd <https://facebook.github.io/zstd/>`_, 
+`iconv <https://www.gnu.org/software/libiconv/>`_, etc. It would be nice to collect them all, regardless of options applied to the **boost**.
+
+.. code-block:: bash
+
+    $ conan install . -g deploy
+
+This helps to collect all the dependencies into a single place, moving them out of the conan cache directory.
+
+.. _using_json_generator:
+
+Using json generator
+--------------------
+
+The more advanced approach is to use the :ref:`json generator <json_generator>`. First off, invoke the conan to generate the ``conanbuildinfo.json`` 
+file containing all required information about dependencies:
+
+.. code-block:: bash
+
+    $ conan install . -g json
+
+The file produced is fully machine-readable and could be used by scripts to prepare the distribution.
+
+.. code-block:: python
+
+        import os
+        import json
+
+        data = json.load(open("conanbuildinfo.json"))
+
+        dep_lib_dirs = dict()
+        dep_bin_dirs = dict()
+
+        for dep in data["dependencies"]:
+            root = dep["rootpath"]
+            lib_paths = dep["lib_paths"]
+            bin_paths = dep["bin_paths"]
+
+            dep_lib_dirs = dict()
+            dep_bin_dirs = dict()
+
+            for lib_path in lib_paths:
+                if os.listdir(lib_path):
+                    lib_dir = os.path.relpath(lib_path, root)
+                    dep_lib_dirs[lib_path] = lib_dir
+            for bin_path in bin_paths:
+                if os.listdir(bin_path):
+                    bin_dir = os.path.relpath(bin_path, root)
+                    dep_bin_dirs[bin_path] = bin_dir
+
+Then, the information collected might be used to copy the required files into the destination:
+
+.. code-block:: python
+
+    from distutils.dir_util import copy_tree
+
+    if not os.path.isdir(destination):
+        os.makedirs(destination)
+    for src_lib_dir, dst_bin_dir in dep_lib_dirs.items():
+        copy_tree(src_lib_dir, os.path.join(destination, dst_lib_dir))
+    for src_bin_dir, dst_bin_dir in dep_bin_dirs.items():
+        copy_tree(src_bin_dir, os.path.join(destination, dst_bin_dir))
+
+The advantage over the ``deploy`` generator is fine-grained control: here we copy only binaries and libraries.
+It's also could be easily modified to apply some sort of filtering (e.g. to copy only shared libraries, 
```

Comment:
I think this is too much. Remove this second example. If the contents are too big people won't read it.

Reply:
sure

Reply:
removed